src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -401,7 +401,23 @@ await _repositoryCloneManager.PrepareCloneAsync(
         }
         catch (NotFoundException)
         {
-            // If target branch does not exist, we create it off of the base branch
+            try
+            {
+                // If target branch does not exist, we create it off of the base branch
+                targetRepo = await _repositoryCloneManager.PrepareCloneAsync(
+                    mapping,
+                    remotes,
+                    [targetBranch],
+                    targetBranch,
+                    ShouldResetVmr,
+                    cancellationToken);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to find branch {branch} in {uri}", targetBranch, string.Join(", ", remotes));
```

Comment:
this will log 2 exceptions in application insights, because we use the `logger.LogError` with the exception

Reply:
It will log this one. Where's the second? `TargetBranchNotFoundException` does not get logged.

Reply:
won't that one fall through to the `WorkItemProcessor` and get logged there? we'll have two at that point

Reply:
ahh my bad, sorry